interactions/api/events/discord.py

```diff
@@ -751,3 +756,46 @@ class GuildAuditLogEntryCreate(GuildEvent):
 
     audit_log_entry: interactions.models.AuditLogEntry = attrs.field(repr=False)
     """The audit log entry object"""
+
+
+@attrs.define(eq=False, order=False, hash=False, kw_only=False)
+class GuildScheduledEventCreate(BaseEvent):
+    """Dispatched when scheduled event is created"""
+
+    scheduled_event: interactions.models.ScheduledEvent
+    """The scheduled event object"""
+
+
+@attrs.define(eq=False, order=False, hash=False, kw_only=False)
+class GuildScheduledEventUpdate(BaseEvent):
+    """Dispatched when scheduled event is updated"""
+
+    before: Absent[interactions.models.ScheduledEvent]
+    """The scheduled event before this event was created"""
+    after: interactions.models.ScheduledEvent
+    """The scheduled event after this event was created"""
+
+
+@attrs.define(eq=False, order=False, hash=False, kw_only=False)
+class GuildScheduledEventDelete(GuildScheduledEventCreate):
+    """Dispatched when scheduled event is deleted"""
+
+
+@attrs.define(eq=False, order=False, hash=False, kw_only=False)
+class GuildScheduledEventUserAdd(GuildEvent):
+    """Dispatched when scheduled event is created"""
+
+    scheduled_event: Optional[interactions.models.ScheduledEvent]
+    """The scheduled event object if cached"""
+    user: "User"
+    """The user that has been added/removed from scheduled event"""
```

Comment:
Why is this not marked optional? This gets the user from its cache, which has a decent chance of not having anything and so returning `None`. I don't believe interactions.py caches any user object right before either.

Reply:
Yeah you're right. I just missed it. Will resolve tomorrow